### PR TITLE
Projector enhancements: more futuristic Look&Feel

### PR DESCRIPTION
**Pull request by TheSilkMiner**
## Description
This PR makes the projector emit light and render a translucent cone when working.

## Extended description
**So, what should this PR do?**
As I previously said, the goal of this PR is giving the projector a more futuristic look, adding various elements commonly associated to holographic projectors.
**Which elements?**
First of all, the blue cone: every projector usually displays either a cone or a box where the object is then rendered. This behaviour is now applicable to the Height Map Projector. Another element is the luminance. Projecting machines usually emit light when displaying something: this is an indicator of their correct and working status. With this PR the same effect applies to OpenBlocks' one.
**Wait a second: this allows the projector to emit light and the branch name is `lights-out`?**
Yep, that's right. Plus that name references [something](https://en.wikipedia.org/wiki/Lights_Out_%282016_film%29)...
**Anyway, that seems nice, can you show me some pictures?**
Yes of course! Here you are [the default version](http://i.imgur.com/oUvYaMf.png) and [the version with grids](http://i.imgur.com/TxPtI8e.png)!
**I noticed that the cone doesn't show some lines. Usually holograms are rendered on a grid: why isn't it present?**
It **is** present, just not enabled by default. This because the grid texture is a little bit pixelated and the corners do not match up perfectly. The cause of this is simply some calculation on the texture side: on a 64 pixel texture, 0.25 corresponds to 16 pixels, but the same does not apply in-game.
If you want to enable the grid anyway, you can do so by setting the value of `renderHolographicGrid` to `true`.
**What about the colour changes? It seems that at night the cone is darker than during the day**
Yep, that's true. This happens because the way we are using to render the cone (`ISBRH`) doesn't allow the modder to disable the brightness effect for that specific render.
You can force the renderer to use a certain brightness value, though. You can do this by setting the value of the option `brightness` to an integer between `0` and `255`. Remember that the lower the value is, the greener the colour the cone will turn into. You can always set the value back to `-1` if you are not satisfied of this behaviour.
**The concept may be nice, but I don't like it. Can I turn it off?**
Of course you can! In OpenBlocks' configuration file, under the `projector` category, you can find various options. The one you are looking for is `renderHolographicCone`. If you set it to `false`, then the cone won't render.
**No no no, I mean the entire concept.**
I'm sorry you don't like it, but that's why I added some more options. Setting `lightLevel` will increase or diminish the light level of the projector, while turning `lightUpWhenWorking` to `false` will disable the entire switching to the working status.
**Wait, why can't I disable the entire block?**
If you want you can: the id is `projector.working`. Disabling the behaviour through the specified configuration options is much better, though. In fact, this will keep the working block state and switch to the previous one when updated, without issues. Deleting the block will erase all the data contained in it, removing all the Height Maps currently inserted.
**Okay, that's it. Thank you for your explanation!**
You're welcome. And thanks for listening to me until this point.

## Full features list
- New working projector block
  - The non-working projector will turn in it whenever a map is placed inside it (configurable)
  - Lights up the area around it (light level configurable)
  - Renders a holographic cone (configurable)
    - Brightness value is configurable
    - Renders a holographic grid on it (configurable, disabled by default)
- Removal of unobtainable blocks from NEI view

### CODE REVIEW

**Comment by boq** [comment]
Few comments:
- What's up with all those absolute class names? Please use imports.
- Replacing TE is risky and breaks few things (like ComputerCraft integration). I strongly suggest using `TileEntity.shouldRefresh` instead.
- Since it's static element, I think ISBRH is more suitable than TESR. Also fixes any transparency problems and is faster.
- It looks like rendering of this block will be scrapped in 1.8 and replaced with proper model (both static and animated parts). It's not an 1.7.10 issue though...

File: src/main/java/openblocks/NEIOpenBlocksConfig.java [hunk 1]
```diff
@@ -34,4 +42,25 @@ public String getVersion() {
 		return "0.0";
 	}
 
+	private void API$hideItem(final ItemStack stack) {
+		try {
+			// Cache method in order to be more performing.
```
Comment (boq) [inline src/main/java/openblocks/NEIOpenBlocksConfig.java]: Needless comments

File: src/main/java/openblocks/NEIOpenBlocksConfig.java [hunk 2]
```diff
@@ -34,4 +42,25 @@ public String getVersion() {
 		return "0.0";
 	}
 
+	private void API$hideItem(final ItemStack stack) {
```
Comment (boq) [inline src/main/java/openblocks/NEIOpenBlocksConfig.java]: Why '$'?

**Comment by boq** [comment]
Hmm, recent changes (imports) broke it. If it's ready, can you align and squeeze commits?

File: src/main/resources/mcmod.info [hunk 3]
```diff
@@ -1,20 +1,33 @@
...
+      ],
+      "dependencies": [
+      ],
+      "dependants": [
+      ],
+      "useDependencyInformation": "false"
+    }
   ]
-}]
+}
```
Comment (boq) [inline src/main/resources/mcmod.info]: Could be split to different PR.

File: src/main/java/openblocks/NEIOpenBlocksConfig.java (hunk 2 above)
Comment (TheSilkMiner) [inline src/main/java/openblocks/NEIOpenBlocksConfig.java]: To split the class name from the method name.
The name is supposed to remember the call to the actual `API.hideItem()` method. Since using a dot is forbidden, I decided to go with a `$`, because it is rarely used and (usually) makes the separation visible.

File: src/main/java/openblocks/client/renderer/block/BlockWorkingProjectorRenderer.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+public class BlockWorkingProjectorRenderer implements IBlockRenderer<BlockWorkingProjector> {
+
+	public static int renderPass = 0;
+
+	private static final float TO_BLOCK_CENTRE = 0.5F;
+	private static final int BRIGHTNESS_LEVEL_MAX = 255;
+	private static final int BRIGHTNESS_LEVEL_MIN = 0;
+	private static final int BRIGHTNESS_LEVEL_DEF = -1;
+
+	private static final float CONE_BEGIN = Config.experimentalFeatures? 0F : 0.25F;
```
Comment (TheSilkMiner) [inline src/main/java/openblocks/client/renderer/block/BlockWorkingProjectorRenderer.java]: Should this be extracted to a projector-related configuration option?

Comment (boq) [inline src/main/java/openblocks/client/renderer/block/BlockWorkingProjectorRenderer.java]: Definitely. Config values are cheap, no point in clumping (possible) multiple functionalities in one.
